plat/kvm/x86/multiboot2.c

```diff
@@ -0,0 +1,188 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/essentials.h>
+#include <uk/arch/limits.h>
+#include <uk/arch/types.h>
+#include <uk/arch/paging.h>
+#include <uk/plat/bootstrap.h>
+#include <uk/plat/common/bootinfo.h>
+#include <uk/plat/common/lcpu.h>
+#include <uk/plat/common/memory.h>
+#include <uk/plat/common/sections.h>
+#include <uk/reloc.h>
+#include <kvm-x86/multiboot2.h>
+
+#include <errno.h>
+#include <string.h>
+
+#define multiboot2_crash(msg, rc)	ukplat_crash()
+
+#define MULTIBOOT_TAG_ALIGN_UP(x)	ALIGN_UP((x), MULTIBOOT_TAG_ALIGN)
+
+void _ukplat_entry(struct lcpu *lcpu, struct ukplat_bootinfo *bi);
+
+static inline void mrd_insert(struct ukplat_bootinfo *bi,
+			      const struct ukplat_memregion_desc *mrd)
+{
+	int rc;
+
+	if (unlikely(mrd->len == 0))
+		return;
+
+	rc = ukplat_memregion_list_insert(&bi->mrds, mrd);
+	if (unlikely(rc < 0))
+		multiboot2_crash("Cannot insert bootinfo memory region", rc);
+}
+
+void handle_mmap_tag(struct ukplat_memregion_desc mrd,
+		     struct ukplat_bootinfo *bi, struct multiboot_tag *tag)
+{
+	__sz i, mmap_total_sz, mmap_entry_cnt;
```

Comment:
:muscle: :christmas_tree:

Reply:
You mean for the function parameters as well or have I misunderstood how it should be done for the declarations starting from line 44?

Reply:
Ah no lol just appreciating that you adopted the style :laughing:

Reply:
Yeah, it's much better this way. Looking back on it, I don't know how I didn't do it like this before :)

Reply:
Eh, it's just a matter of style :smile:. Unless mandated by the coding style policy, feel free to do things your own way. I just like this style visually. Got it from the Linux Kernel as they have started adopting this on the mailing lists as well.